cluster-autoscaler/cloudprovider/gce/templates_test.go

```diff
@@ -158,16 +182,25 @@ func TestBuildNodeFromTemplateSetsResources(t *testing.T) {
 						{
 							Boot: true,
 							InitializeParams: &gce.AttachedDiskInitializeParams{
-								DiskSizeGb: tc.physicalEphemeralStorage,
+								DiskSizeGb: bootDiskSizeGiB,
 							},
 						},
 					},
 				},
 			}
+			for i := int64(0); i < tc.attachedLocalSSDCount; i++ {
+				template.Properties.Disks = append(template.Properties.Disks, &gce.AttachedDisk{
+					Type: "SCRATCH",
+					InitializeParams: &gce.AttachedDiskInitializeParams{
+						DiskType:   "local-ssd",
+						DiskSizeGb: localSSDSizeGiB,
+					},
+				})
+			}
 			if tc.kubeEnv != "" {
 				template.Properties.Metadata.Items = []*gce.MetadataItems{{Key: "kube-env", Value: &tc.kubeEnv}}
 			}
-			node, err := tb.BuildNodeFromTemplate(mig, template, tc.physicalCpu, tc.physicalMemory, tc.pods)
+			node, err := tb.BuildNodeFromTemplate(mig, template, physicalCpu, physicalMemory, tc.pods)
```

Comment:
1) The same value is used for all tests.
2) These values aren't variables of the system under test. As explained in the comment that I added, the test itself computes expected CPU/memory capacity/allocatable, so it doesn't make sense to use different values for different test cases.

Reply:
I was trying to wrap my head around what this test actually tests, since it's calling submethods BuildCapacity and CalculateAllocatable in the test itself. It turns out physicalCpu and physicalMemory are test invariants: changing their values doesn't affect the test results, because they are used to compute the expected results as well.

That being said, someone could add logic in BuildNodeFromTemplate, outside of BuildCapacity and CalculateAllocatable, that would use physicalCpu and physicalMemory as variables, in which case they wouldn't be invariants anymore, but that's not currently the case.

Anyway, I've dropped my changes in this regard because I don't want this PR to be blocked because of my attempt to clarify test boundaries. I'm happy to let someone else fix that later, if anyone cares.